### DISCUSSION TITLE
Public form breaks with illegal doc ID

### DISCUSSION THREAD

**Discussion by juliansworkshop** [Budibase/budibase#2658] [Help]
have a public app with public data when its run without logging in I'm seeing this error it runs fine when logged in

bbapps                | ERROR [1631999149346] (27 on 3cb98694c922): Document id must be a string
bbapps                |     illegal_docid: Document id must be a string
bbapps                |         at Object.throw (/app/node_modules/koa/lib/context.js:97:11)
bbapps                |         at /app/dist/api/controllers/row/index.js:48:18
bbapps                |         at Generator.throw (<anonymous>)
bbapps                |         at rejected (/app/dist/api/controllers/row/index.js:6:65)
bbapps                |         at runMicrotasks (<anonymous>)
bbapps                |         at processTicksAndRejections (internal/process/task_queues.js:97:5)
bbapps                | Trace: illegal_docid: Document id must be a string
bbapps                |     at Object.throw (/app/node_modules/koa/lib/context.js:97:11)
bbapps                |     at /app/dist/api/controllers/row/index.js:48:18
bbapps                |     at Generator.throw (<anonymous>)
bbapps                |     at rejected (/app/dist/api/controllers/row/index.js:6:65)
bbapps                |     at runMicrotasks (<anonymous>)
bbapps                |     at processTicksAndRejections (internal/process/task_queues.js:97:5) {
bbapps                |   error: 'illegal_docid',
bbapps                |   reason: 'Document id must be a string'
bbapps                | }

![image](https://user-images.githubusercontent.com/54377525/133926471-456460ee-90ff-40f3-8a7d-e3d81fd998d0.png)
![image](https://user-images.githubusercontent.com/54377525/133926275-19e5c552-46ce-45ee-8eae-0a2f856cdec4.png)

I've rebuilt the app and data to make sure its replicateable

**Comment by mjashanks** [comment]
Hi @juliansworkshop - I am able to reproduce this. Although, I'm surprised because we run some public forms, that are not broken! Something must have changed recently to break this.

I have raised this here: https://github.com/Budibase/budibase/issues/2665 - we will fix this very soon.

**Reply by juliansworkshop** [reply]
I also run some that arn't broken I think its due to the way its updating from a repeater

**Reply by juliansworkshop** [reply]
If it helps I tried the same thing with a SQL server I get no error but it doesn't save anything.

**Reply by mjashanks** [reply]
Hmm thats very interesting...thanks.

**Reply by mjashanks** [reply]
@mslourens just pointed out that this is caused by the CreatedBy / UpdatedBy Auto columns: https://github.com/Budibase/budibase/issues/2665#issuecomment-922866130

I removed the columns, which fixed the issue. I still think that this is a bug, but hopefully this is a workaround for you.

**Reply by juliansworkshop** [reply]
Yea updated by seems to have fixed it in this case Just checked the other apps I must've deselected the created by for them. Guess the bug is that there's no unknown user for that to link to.
